server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -745,9 +746,8 @@ public void deleteStaleSegments(int lastNMetadataFilesToKeep) throws IOException
             return;
         }
 
-        List<String> metadataFilesEligibleToDelete = sortedMetadataFileList.subList(
-            lastNMetadataFilesToKeep,
-            sortedMetadataFileList.size()
+        List<String> metadataFilesEligibleToDelete = new ArrayList<>(
```

Comment:
Are the changes in this file something that should be done separately from this PR? (especially since this change might not be backported in the short term)

Reply:
Good question, I was consistently hitting this issue only with my changes, and haven't seen an issue created.  I can follow up separately to backport the fix for the concurrent modification exception + test logging.  What do you think?

Reply:
I'll break this out into its own change since its likely causing issues elsewhere

Reply:
https://github.com/opensearch-project/OpenSearch/pull/12328